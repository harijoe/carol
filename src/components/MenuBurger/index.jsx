import React, { Component } from 'react'
import { Link } from 'react-router'
import LinkPro from '../../containers/pro/LinkPro'
import Menu from '../Menu'

class MenuBurger extends Component {
  constructor(props) {
    super(props)

    this.state = {
      country: props.country
    }
  }

  render() {
    return (
      <Menu label="Burger menu">
        <ul>
          <li><Link to="/">Home</Link></li>
          <li><Link to="/search-pro">Trouver un artisan</Link></li>
          <li><Link to="*">Ressources pour mes travaux</Link></li>
          <li><LinkPro country={this.props.country} /></li>
        </ul>
      </Menu>
    )
  }
}

MenuBurger.propTypes = {
  country: React.PropTypes.string
}

export default MenuBurger

