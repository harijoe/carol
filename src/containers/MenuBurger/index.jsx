import React from 'react'
import { Link } from 'react-router'
import { FormattedMessage } from 'react-intl'
import LinkPro from '../../containers/pro/LinkPro'
import Menu from '../Menu'

export default function MenuBurger () {
  return (
    <Menu label="burger_menu">
      <ul>
        <li>
          <Link to="/">
            <FormattedMessage id="home" />
          </Link>
        </li>
        <li>
          <Link to="/search-pro">
            <FormattedMessage id="pro.find_pro" />
          </Link>
        </li>
        <li>
          <Link to="*">
            <FormattedMessage id="pro.resource" />
          </Link>
        </li>
        <li>
          <Link to="/help">
            <FormattedMessage id="help" />
          </Link>
        </li>
        <li><LinkPro /></li>
      </ul>
    </Menu>
  )
}
