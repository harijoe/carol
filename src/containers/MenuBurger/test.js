import chai from 'chai'
import rewire from 'rewire'
import { shallow } from 'enzyme'
import React from 'react'
import { fromJS } from 'immutable'

const expect = chai.expect
const MenuBurgerConnect = rewire("./")
const MenuBurger = MenuBurgerConnect.__get__('MenuBurger')

describe('<MenuBurger />', () => {
  it('should have <Menu country={country: fromJS({ countryCode: "GB"}) } /> with a list of 4 items', () => {
    const wrapper = shallow(<MenuBurger country={fromJS({ countryCode: 'GB'})} />)

    expect(wrapper.find('Connect(Menu)')).to.have.length(1)
    expect(wrapper.find('Connect(Menu) ul li')).to.have.length(4)
  })
})
