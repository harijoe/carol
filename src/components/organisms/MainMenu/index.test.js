import React from 'react'
import { shallow } from 'enzyme'

import MenuBurger from './index'
import intlMock from '../../../../test/intlMock'

const wrap = (props = {}) => shallow(intlMock(<MenuBurger linkPro="https://www.quotatispro.fr/" {...props} />))

it('renders MenuBurger', () => {
  expect(wrap()).toMatchSnapshot()
})
