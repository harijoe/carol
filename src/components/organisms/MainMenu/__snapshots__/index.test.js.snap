// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MenuBurger 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider
    locale="en"
    messages={Object {}}
  >
    <injectScrollWrapper
      linkPro="https://www.quotatispro.fr/"
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "linkPro": "https://www.quotatispro.fr/",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "linkPro": "https://www.quotatispro.fr/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
